File: 백준/Silver/2847. 게임을 만든 동준이/게임을 만든 동준이.cc
#include <iostream>
#include <vector>
#include <algorithm>
#include <queue>


using namespace std;

int main() {
    ios_base::sync_with_stdio(false);
    cin.tie(nullptr);

    int n, tmp;
    vector<int> scores;
    cin >> n;
    for (int i = 0; i < n; i++) {
        cin >> tmp;
        scores.push_back(tmp);
    }

    int cnt = 0;
    for (int i = n - 1; i > 0; i--) {
        if (scores[i - 1] >= scores[i]) {
            int differ = scores[i - 1] - scores[i] + 1;
            scores[i - 1] -= differ;
            cnt += differ;
        }
    }

    cout << cnt;

    return 0;
}

File: 백준/Silver/2847. 게임을 만든 동준이/게임을 만든 동준이.py


def main():
    n = int(input())
    scores = []

    for i in range(n):
        scores.append(int(input()))

    cnt = 0
    for i in range(n - 1, 0, -1):
        if scores[i - 1] >= scores[i]:
            differ = scores[i - 1] - scores[i] + 1 
            scores[i - 1] -= differ
            cnt += differ
    
    print(cnt)


if __name__ == "__main__":
    main()